patches/ios-web_view-internal-cwv_web_view_configuration.mm.patch

```diff
@@ -0,0 +1,98 @@
+diff --git a/ios/web_view/internal/cwv_web_view_configuration.mm b/ios/web_view/internal/cwv_web_view_configuration.mm
+index db4c4bd18207dbfe22c05cb6da2fdb94dced9393..7110894808fa2b710879720ae95e3d5c4130dd6c 100644
+--- a/ios/web_view/internal/cwv_web_view_configuration.mm
++++ b/ios/web_view/internal/cwv_web_view_configuration.mm
+@@ -5,13 +5,14 @@
+ #import <memory>
+ 
+ #import "base/threading/thread_restrictions.h"
+-#import "components/affiliations/core/browser/affiliation_service.h"
+ #import "components/keyed_service/core/service_access_type.h"
+ #import "components/keyed_service/ios/browser_state_dependency_manager.h"
+ #import "components/password_manager/core/browser/leak_detection/bulk_leak_check_service_interface.h"
+ #import "components/password_manager/core/browser/password_store/password_store_interface.h"
+ #import "components/sync/service/sync_service.h"
+-#import "ios/web_view/internal/affiliations/web_view_affiliation_service_factory.h"
++#import "ios/chrome/browser/shared/model/application_context/application_context.h"
++#import "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
++#import "ios/chrome/browser/shared/model/browser_state/chrome_browser_state_manager.h"
+ #import "ios/web_view/internal/app/application_context.h"
+ #import "ios/web_view/internal/autofill/cwv_autofill_data_manager_internal.h"
+ #import "ios/web_view/internal/autofill/web_view_personal_data_manager_factory.h"
+@@ -38,7 +39,7 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ 
+ @interface CWVWebViewConfiguration () {
+   // The BrowserState for this configuration.
+-  std::unique_ptr<ios_web_view::WebViewBrowserState> _browserState;
++  ios_web_view::WebViewBrowserState* _browserState;
+ 
+   // Holds all CWVWebViews created with this class. Weak references.
+   NSHashTable* _webViews;
+@@ -82,10 +83,13 @@ NSHashTable<CWVWebViewConfiguration*>* gNonPersistentConfigurations = nil;
+ + (instancetype)defaultConfiguration {
+   static dispatch_once_t onceToken;
+   dispatch_once(&onceToken, ^{
+-    auto browserState = std::make_unique<ios_web_view::WebViewBrowserState>(
+-        /* off_the_record = */ false);
++    ios::ChromeBrowserStateManager* browserStateManager =
++        GetApplicationContext()->GetChromeBrowserStateManager();
++    ChromeBrowserState* browserState =
++        browserStateManager->GetLastUsedBrowserStateDeprecatedDoNotUse();
```

Comment:
this is definitely not safe to use here and we shouldn't be using `GetLastUsedBrowserStateDeprecatedDoNotUse` in any new places
```
// states was used last. Do *not* use it as a singleton getter to fetch "the"
  // browser state. Always assume there could be multiple browser states and
  // use GetLoadedBrowserStates() instead.
```

Reply:
also how are we handling regular vs incognito here? We need both for proper handling of keyed services

Reply:
We'd use `+[CWVWebViewConfiguration nonPersistentConfiguration]` not `defaultConfiguration` for incognito, we specify that in `TabManager`

Reply:
> this is definitely not safe to use here and we shouldn't be using `GetLastUsedBrowserStateDeprecatedDoNotUse` in any new places

We'd address this I think with https://github.com/brave/brave-browser/issues/38419